src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/MermaidKernelExtensionTests.cs

```diff
@@ -0,0 +1,136 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Assent;
+
+using FluentAssertions;
+
+using Microsoft.DotNet.Interactive.CSharp;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.Formatting;
+using Microsoft.DotNet.Interactive.Tests.Utility;
+
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.ExtensionLab.Tests
+{
+    public class MermaidKernelExtensionTests : IDisposable
+    {
+        private readonly Configuration _configuration;
+
+        public MermaidKernelExtensionTests()
+        {
+            _configuration = new Configuration()
+                .UsingExtension("txt")
+                .SetInteractive(Debugger.IsAttached);
+        }
+
+        [Fact]
+        public async Task adds_mermaid_kernel()
+        {
+            using var kernel = new CompositeKernel
+            {
+                new CSharpKernel().UseNugetDirective(),
+            };
+
+            var extension = new MermaidKernelExtension();
+
+            await extension.OnLoadAsync(kernel);
+
+            kernel.ChildKernels
+                .Should()
+                .ContainSingle(k => k is MermaidKernel);
+        }
+
+        [Fact]
+        public async Task register_formatters()
+        {
+            using var kernel = new CompositeKernel
+            {
+                new CSharpKernel().UseNugetDirective(),
+            };
+
+            var extension = new MermaidKernelExtension();
+
+            await extension.OnLoadAsync(kernel);
+
+            var formatted = new MermaidMarkdown(@"
+    graph TD
+    A[Client] --> B[Load Balancer]
+    B --> C[Server1]
+    B --> D[Server2]").ToDisplayString(HtmlFormatter.MimeType);
+
+            this.Assent(formatted.FixedGuid());
+        }
+
+
+        [Fact]
+        public async Task mermaid_kernel_handles_SubmitCode()
+        {
+            using var kernel = new CompositeKernel
+            {
+                new CSharpKernel().UseNugetDirective(),
+            };
+
+            var extension = new MermaidKernelExtension();
+
+            await extension.OnLoadAsync(kernel);
+
+            var result = await kernel.SubmitCodeAsync(@"
+#!mermaid
+    graph TD
+    A[Client] --> B[Load Balancer]
+    B --> C[Server1]
+    B --> D[Server2]
+");
+
+            var events = result.KernelEvents.ToSubscribedList();
+
+            var formattedData = events
+                .OfType<DisplayedValueProduced>()
+                .Single()
+                .FormattedValues
+                .Single(fm => fm.MimeType == HtmlFormatter.MimeType)
+                .Value;
+
+            this.Assent(formattedData.FixedGuid(), _configuration);
+        }
+
+        [Fact]
+        public void can_generate_class_diagram_from_type()
+        {
+            var diagram = typeof(CompositeKernel).ToClassDiagram();
```

Comment:
It might be a good idea to generate this from something more stable.

That said, this also looks like it might be a very nice way to built an assertion that the public API hasn't changed, once a type is stable, e.g. https://github.com/dotnet/command-line-api/issues/1242#issuecomment-811531652.

Reply:
sure

Reply:
We used to have a tool that did something like this, back when developing Windows Vista (or was it XP?) ... to ensure the managed APIs were unchanged between builds.  (In part it was to prevent cross-branch build breaks.)  So, I'm pretty sure a tool already exists for ensuring exposed APIs are unchanged.... might not need to write one is what I mean.

Reply:
We we don't need to write one, just write a few lines of code in a test plus [Assent](https://www.nuget.org/packages/Assent/) or [ApprovalTests](https://www.nuget.org/packages/approvaltests/).